test/test_db.py

```diff
@@ -38,41 +38,33 @@ def teardown_function(function):
     os.remove(db_filename)
 
 
-def test_get_nick_id(db):
+@pytest.mark.parametrize('name, slug, variant', (
+    # Check case insensitive with ASCII only
+    ('Embolalia', 'embolalia', 'eMBOLALIA'),
+    # Ensures case conversion is handled properly
+    ('[][]', '{}{}', '[}{]'),
+    # Unicode, just in case
+    ('EmbölaliÅ', 'embölaliÅ', 'EMBöLALIÅ'),
+))
```

Comment:
Doesn't switching to this remove the verification that different nicks (that aren't equivalent once casemapped) do indeed get different IDs? I think that's why the test used a loop structure before.

Reply:
The previous version thought that they were the same, because it used `str.lower()`, which is not ascii, rfc1459, nor rfc1459-strict. 😁 

This is the part that is "breaking change", which should affect server that allows UTF-8 nicknames and channel names, which will be solved by a future PR trying to implement something like UTF8MAPPING and what is suggested in RFC 7564 (and its upgraded version RFC 8264).

Reply:
Seems I didn't quite get my question across, probably due to attaching this comment to lines in the wrong place.

I meant that there doesn't seem to be any equivalent in the new test code below that matches this old check:

```py
    # Check that each nick ended up with a different id
    assert len(set(test[0] for test in tests)) == len(tests)
```

Each parametrized version of the test is going to start with a fresh `db` object, right? So unless I've missed something (always possible 😅), the test has been downgraded to only making sure that `get_nick_id(somenick, create=True)` retrieves an ID, rather than making sure it creates a _new_ ID when expected.

Put another way, nothing about this new test that I can see makes sure that e.g. the ID for the nick `reblochon`, when seen for the first time, is different from the ID for `brie` (also seen for the first time).

Reply:
Bonus point: I've added some comment about why `create=True`. However, I didn't comment on `create=False`, because I'm not documenting the past here, only what *is* as the moment.